realm/realm-library/src/main/java/io/realm/internal/android/JsonUtils.java

```diff
@@ -41,15 +43,18 @@ public static Date stringToDate(String date) {
         if (matcher.find()) {
             String dateMatch = matcher.group(1);
             return new Date(Long.parseLong(dateMatch));
-        } else {
-            return new Date(Long.parseLong(date));
         }
+        try {
+            return ISO8601Utils.parse(date, new ParsePosition(0));
```

Comment:
I could be tempted to say we could remove the `ParsePosition` class as it doesn't make sense in our context, but then it would be hard to keep our version of `ISO8601Utils` up to date. At least saving `ParsePosition` in a static variable will reduce objects allocated when parsing large amounts of Json.